xla/stream_executor/cuda/cuda_command_buffer.cc

```diff
@@ -585,15 +715,100 @@ absl::Status CudaCommandBuffer::Trace(
 #endif
 }
 
-absl::Status CudaCommandBuffer::SetNodeExecutionEnabled(
-    GraphNodeHandle node_handle, bool enabled) {
-  // Node is enabled if value != 0, otherwise the node is disabled.
-  unsigned value = enabled ? 1 : 0;
-  VLOG(2) << "Set CUDA executable graph " << exec_ << " node " << node_handle
-          << " enabled flag to " << value;
-  return cuda::ToStatus(
-      cuGraphNodeSetEnabled(exec_, ToCudaGraphHandle(node_handle), value),
-      "Failed to set CUDA graph node enabled flag");
+/*static*/ int64_t CudaCommandBuffer::NotifyExecCreated() {
+  alive_execs.fetch_add(1, std::memory_order_relaxed);
+  return allocated_execs.fetch_add(1, std::memory_order_relaxed);
+}
+
+/*static*/ int64_t CudaCommandBuffer::NotifyExecDestroyed() {
+  DCHECK_GE(alive_execs.load(std::memory_order_relaxed), 1);
+  return alive_execs.fetch_sub(1, std::memory_order_relaxed) - 1;
+}
+
+/*static*/ int64_t CudaCommandBuffer::AliveExecs() {
+  return alive_execs.load(std::memory_order_relaxed);
+}
+
+absl::Status CudaCommandBuffer::Finalize() {
+  TF_RETURN_IF_ERROR(PrepareFinalization());
+
+  // Maybe dump created CUDA graph to a dot file for debugging.
+  if (VLOG_IS_ON(10) && mode() == Mode::kPrimary) {
+    std::string path = tsl::io::GetTempFilename(/*extension=*/"dot");
+    TF_RETURN_IF_ERROR(WriteGraphToDotFile(path));
+    if (VLOG_IS_ON(100)) {
+      std::string dot_file_contents;
+      TF_RETURN_IF_ERROR(
+          tsl::ReadFileToString(tsl::Env::Default(), path, &dot_file_contents));
+      VLOG(100) << "Contents of " << path << " is:\n" << dot_file_contents;
+    }
+  }
+
+  // Collect number of nodes and conditionals for logging below.
+  size_t num_nodes = 0, num_cond_cmd_buffers = 0;
+
+  if (mode() == Mode::kPrimary) {
+    uint64_t start_nanos = tsl::Env::Default()->NowNanos();
+
+    if (exec_ == nullptr) {
+      // If this is the first time we finalize command buffer after
+      // construction, we need to instantiate it to an executable graph.
+      auto instantiated = InstantiateGraph();
+
+      if (instantiated.code() == absl::StatusCode::kResourceExhausted) {
+        return absl::ResourceExhaustedError(absl::StrFormat(
+            "Underlying backend ran out of memory trying to instantiate "
+            "graph "
```

Comment:
formatting is off a little

Reply:
This is the clang format run results,  just keep it like this?  Afraid of manual change can not pass clang format error report.